ui/pages/bridge/transaction-details/step-progress-bar-item.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { StatusTypes } from '../../../../shared/types/bridge-status';
+import {
+  Icon,
+  IconName,
+  IconSize,
+} from '../../../components/component-library';
+import { IconColor } from '../../../helpers/constants/design-system';
+import HollowCircle from './hollow-circle';
+import PulsingCircle from './pulsing-circle';
+
+const ICON_SIZE = IconSize.Xs;
+
+const VerticalLine = ({ color }: { color: IconColor }) => (
+  <div
+    style={{
+      height: '60px',
+      marginTop: '-1rem',
+      marginBottom: '-1rem',
+      width: '1px',
+      backgroundColor: `var(--color-${color})`,
+      zIndex: 0.1,
+    }}
+  />
+);
+
+type StepsProgressBarItemProps = {
+  stepStatus: StatusTypes | null;
+  isLastItem: boolean;
+  isEdgeComplete: boolean;
+  children: React.ReactNode;
+};
+
+/**
+ * Renders the steps in the Bridge Transaction Details page
+ *
+ * @param options
+ * @param options.stepStatus - The status of the step
+ * @param options.isLastItem - Whether the step is the last item
+ * @param options.isEdgeComplete - Whether the edge is complete
+ * @param options.children - The description of the step to be rendered
+ */
+export default function StepProgressBarItem({
+  stepStatus,
+  isLastItem,
+  isEdgeComplete,
+  children,
+}: StepsProgressBarItemProps) {
+  return (
+    <>
+      {/* Indicator dots */}
+      {stepStatus === null && (
+        <HollowCircle size={ICON_SIZE} color={IconColor.iconMuted} />
+      )}
+      {stepStatus === StatusTypes.PENDING && (
+        <PulsingCircle iconSize={ICON_SIZE} color={IconColor.primaryDefault} />
+      )}
+      {stepStatus === StatusTypes.COMPLETE && (
```

Comment:
Is there a separate ui treatment for failed transactions?

Reply:
Will handle failed txs in a separate PR while we wait for designs